app/controllers/TestUserController.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+class UsersController extends \BaseController {
+
+	/**
+	 * Display a listing of the resource.
+	 * GET /users
+	 *
+	 * @return Response
+	 */
+	public function index()
+	{		
+		$user 		= 	Auth::user();
+		$created 	= 	$user->tasks_created;
+		$completed 	= 	$user->tasks_completed;
+
+		$pTitle		=	Auth::user()->full_name;
+
+		if ($created == "") {
+			$created = 0;
+		}
+
+		if ($completed == "") {
+			$completed = 0;
+		}
+		return View::make('users.index',compact(['user','created','completed','pTitle']));			
+	}
+
+	/**
+	 * Remove the specified resource from storage.
+	 * DELETE /users/{id}
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function destroy($id)
+	{
+		// Delete everything related to the user
+		Task::where('user_id', Auth::id())->delete(); 
+		Credential::where('user_id', Auth::id())->delete();
+		Project::where('user_id', Auth::id())->delete();
+		Client::where('user_id', Auth::id())->delete();		
+		User::where('id', Auth::id())->delete();
+
+		// Logout and redirect back to home page
+		Auth::logout();
+		return Redirect::to('/');
+	}
+
+	/**
+	 * Logout the user from the application
+	 */	
+	public function logout(){
+		Auth::logout();
+		return Redirect::to('/');
+	}
+
+	/**
+	 * Login the user and start a session
+	 */
+	public function login()
+	{				
+		$email 		=	Input::get('email');
+		$password	=	Input::get('password');
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "password" 10 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://172.27.59.109:9000/coding_rules#rule_key=php%3AS1192)